### DISCUSSION TITLE
Song Offset

### DISCUSSION THREAD

**Discussion by THEkatinamicrowave** [CodenameCrew/CodenameEngine#396] [Q&A]
This is about source code btw, not actual settings. I'm trying to fix the source code to work how one would expect it to.

So I've confirmed at this point that my headphones have an offset of exactly 239 ms. Now when I go into options on CNE (and only CNE), 239 makes the notes way slower. For whatever reason, on CNE, my song offset needs to be set to 165 to be accurate. Why? Beats me. I looked in the source code and it seems like it should be converting the songOffset value to milliseconds. I really have no idea. Please help.

**Comment by r6915ee** [comment]
Huh, weird. I don't usually configure with anything related to my audio unless it's for something like volume, so I'm not really an expert on this, but alas. What other engines or games have you at least tested with for this case?

**Reply by THEkatinamicrowave** [reply]
I've tested it with FNF (main game), Psych Engine (latest version), Kade Engine (latest version), and Funky Friday (roblox)

**Reply by THEkatinamicrowave** [reply]
but yeah based on this code, I can clearly tell that it's at least _trying_ to use milliseconds
![image](https://github.com/user-attachments/assets/014c4fef-502e-4645-969c-b78b0b953993)

**Reply by THEkatinamicrowave** [reply]
actually wait lme try this
![image](https://github.com/user-attachments/assets/cae81205-f8cf-4368-848c-c0c940673fbc)

**Reply by THEkatinamicrowave** [reply]
whoops
![image](https://github.com/user-attachments/assets/f2b0fc72-92bf-4be3-90c4-f81a2bbc68af)

**Reply by THEkatinamicrowave** [reply]
I have made a severe and continuous lapse in my judgement
Everything that is supposed to be on beat is now ultra fast

Oh and song offset works properly now

**Reply by THEkatinamicrowave** [reply]
I'll be on tomorrow at 1PM EST

**Reply by r6915ee** [reply]
> I have made a severe and continuous lapse in my judgement Everything that is supposed to be on beat is now ultra fast
> 
> Oh and song offset works properly now

I mean, it's at least good that the offset is now correct. But now there's also the beat speed to consider.

**Reply by THEkatinamicrowave** [reply]
For now I set the code back to what it was before
I'm trying to parse what each variable is referring to and reverse engineer everything
But yeah I have no idea as of now what could be happening with the offset

**Reply by r6915ee** [reply]
> For now I set the code back to what it was before I'm trying to parse what each variable is referring to and reverse engineer everything But yeah I have no idea as of now what could be happening with the offset

I'm not really that familiar with the song-related code specifically, so I'm not exactly sure what each variable means as well. So I don't know how to help on that end.

**Reply by THEkatinamicrowave** [reply]
aight that's fine thanks anyway
this should have gone in Issues instead of Discussions either way
